File: daq/slc/base/src/Connection.cc
/**************************************************************************
 * basf2 (Belle II Analysis Software Framework)                           *
 * Author: The Belle II Collaboration                                     *
 *                                                                        *
 * See git log for contributors and copyright holders.                    *
 * This file is licensed under LGPL-3.0, see LICENSE.md.                  *
 **************************************************************************/
#include "daq/slc/base/Connection.h"

using namespace Belle2;

const Connection Connection::ONLINE(10000, "ONLINE");
const Connection Connection::OFFLINE(20000, "OFFLINE");

Connection::Connection() : Enum()
{

}

Connection::Connection(const Enum& st)
  : Enum(st.getId(), st.getLabel())
{

}

Connection::Connection(const Connection& st)
  : Enum(st.getId(), st.getLabel())
{

}

Connection::Connection(int id, const char* label)
  : Enum(id, label)
{

}

Connection::~Connection()
{

}

const Connection& Connection::operator=(const std::string& msg)
{
  if (msg == ONLINE.getLabel()) *this = ONLINE;
  else if (msg == OFFLINE.getLabel()) *this = OFFLINE;
  else *this = UNKNOWN;
  return *this;
}

const Connection& Connection::operator=(const char* msg)
{
  if (msg != NULL)  *this = std::string(msg);
  else *this = UNKNOWN;
  return *this;
}

const Connection& Connection::operator=(int i)
{
  if (i == ONLINE.getId()) *this = ONLINE;
  else if (i == OFFLINE.getId()) *this = OFFLINE;
  else *this = UNKNOWN;
  return *this;
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/core/Connection.java
package b2daq.core;

public class Connection extends Enum {

	final static public Connection ONLINE = new Connection(10000, "ONLINE"); 
	final static public Connection OFFLINE = new Connection(20000, "OFFLINE"); 
	
	public Connection() {
		
	}
	
	public Connection(int id, String label) {
		super(id, label);
	}

	public Connection(Connection connection) {
		copy(connection);
	}

	public void copy(int id) {
		  if ( id == ONLINE._id ) copy(ONLINE);
		  else if ( id == OFFLINE._id ) copy(OFFLINE);
	}

}
